client/blocks/conversation-follow-button/button.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+class ConversationFollowButton extends React.Component {
+	static propTypes = {
+		isFollowing: PropTypes.bool.isRequired,
+		onFollowToggle: PropTypes.func,
+		tagName: PropTypes.oneOfType( [ PropTypes.string, PropTypes.func ] ),
+	};
+
+	static defaultProps = {
+		isFollowing: false,
+		onFollowToggle: noop,
+		tagName: 'button',
+	};
+
+	toggleFollow = event => {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		if ( this.props.onFollowToggle ) {
+			this.props.onFollowToggle( ! this.props.isFollowing );
+		}
+	};
+
+	render() {
+		const { isFollowing, translate } = this.props;
+		const buttonClasses = [
+			'button',
+			'has-icon',
+			'conversation-follow-button',
+			this.props.className,
+		];
+		const iconSize = 20;
+		const label = isFollowing
+			? translate( 'Following Conversation' )
+			: translate( 'Follow Conversation' );
+
+		if ( this.props.isFollowing ) {
+			buttonClasses.push( 'is-following' );
+		}
+
+		const followingIcon = (
+				<Gridicon key="following" icon="reader-following-conversation" size={ iconSize } />
+			),
+			followIcon = <Gridicon key="follow" icon="reader-follow-conversation" size={ iconSize } />,
+			followLabelElement = (
+				<span key="label" className="conversation-follow-button__label">
+					{ label }
+				</span>
+			);
+
+		return React.createElement(
+			this.props.tagName,
```

Comment:
do we need to be anything but a `button` ever?  if we could get rid of tagName and write this in jsx instead of `React.createElement` that would be much nicer

Reply:
Yep - in the post options menu it gets wrapped in a `PostOptionsMenuItem`. The main follow button has a `tagName` prop for this reason too.

Reply:
why dont we have the `button` within a `PostOptionsMenuItem`? 
will that be difficult to style?  Doing it this way feels to me like its a bit of cross-cutting concerns that should be separate.  I don't want to hold up this PR that and would be happy to chase this separate and update it for both FollowButton and FollowConversationButton in a later PR if there is a clean way to refactor this